### PR TITLE
Updated remediation script of deprecated account

### PR DESCRIPTION
**Pull request by SINIKI**

### CODE REVIEW

File: Scripts/RemediationScripts/Remediate-InvalidAADObjectRoleAssignments.ps1 [hunk 1]
```diff
@@ -215,11 +215,8 @@ function Remove-AzTSInvalidAADAccounts
         if(($ObjectIds | Measure-Object).Count -eq 0)
         {
             # Getting all classic role assignments.
-            $armUri = "https://management.azure.com/subscriptions/$($subscriptionId)/providers/Microsoft.Authorization/classicadministrators?api-version=2015-06-01"
-            $method = "Get"
             $classicAssignments = [ClassicRoleAssignments]::new()
-            $headers = $classicAssignments.GetAuthHeader()
-            $res = $classicAssignments.GetClassicRoleAssignmnets([string] $armUri, [string] $method, [psobject] $headers)
+            $res = $classicAssignments.GetClassicRoleAssignmnets($subscriptionId)
```
Comment (gvaradarajan-msft) [inline Scripts/RemediationScripts/Remediate-InvalidAADObjectRoleAssignments.ps1:219]: Typo in GetClassicRoleAssign**mne**ts.

File: Scripts/RemediationScripts/Remediate-InvalidAADObjectRoleAssignments.ps1 [hunk 2]
```diff
@@ -440,42 +438,51 @@ function Remove-AzTSInvalidAADAccounts
...
-                Remove-AzRoleAssignment $_ 
-                $_ | Select-Object -Property "Scope", "RoleDefinitionName", "ObjectId"    
-            }
-            catch
-            {
-                $isRemoved = $false
-                Write-Host "Not able to remove invalid role assignment. ErrorMessage [$($_)]" -ForegroundColor Red  
+        if (($invalidAADObjectRoleAssignments | Measure-Object).Count -gt 0)
+        {
+
```
Comment (gvaradarajan-msft) [inline Scripts/RemediationScripts/Remediate-InvalidAADObjectRoleAssignments.ps1:443]: Please remove this stray newline.

File: Scripts/RemediationScripts/Remediate-InvalidAADObjectRoleAssignments.ps1 [hunk 3]
```diff
@@ -440,42 +438,51 @@ function Remove-AzTSInvalidAADAccounts
...
                     {
-                        $_ | Select-Object -Property "SignInName", "RoleAssignmentId", "RoleDefinitionName"
+                        $isServiceAdminAccount = $false
+                        if($_.RoleDefinitionName -eq "ServiceAdministrator")
+                        {
+                            $isServiceAdminAccount = $true;
+                        }
+
+                        $classicAssignments = [ClassicRoleAssignments]::new()
+                        $res = $classicAssignments.DeleteClassicRoleAssignmnets($_.RoleAssignmentId, $isServiceAdminAccount)
```
Comment (gvaradarajan-msft) [inline Scripts/RemediationScripts/Remediate-InvalidAADObjectRoleAssignments.ps1:473]: Typo in GetClassicRoleAssign**mne**ts.

File: Scripts/RemediationScripts/Remediate-InvalidAADObjectRoleAssignments.ps1 [hunk 4]
```diff
@@ -543,20 +550,26 @@ class ClassicRoleAssignments
         return($content)
     }
 
-    [PSObject] DeleteClassicRoleAssignmnets([string] $armUri, [string] $method, [psobject] $headers)
+    [PSObject] DeleteClassicRoleAssignmnets([string] $roleAssignmentId, [bool] $isServiceAdminAccount)
```
Comment (gvaradarajan-msft) [inline Scripts/RemediationScripts/Remediate-InvalidAADObjectRoleAssignments.ps1:553]: Typo in DeleteClassicRoleAssign**mne**ts.

**Review by gvaradarajan-msft — CHANGES_REQUESTED** [review]

File: Scripts/RemediationScripts/Remediate-InvalidAADObjectRoleAssignments.ps1 (hunk 3 above)
Comment (SINIKI) [reply Scripts/RemediationScripts/Remediate-InvalidAADObjectRoleAssignments.ps1]: Corrected.

Comment (SINIKI) [reply Scripts/RemediationScripts/Remediate-InvalidAADObjectRoleAssignments.ps1]: Corrected.

File: Scripts/RemediationScripts/Remediate-InvalidAADObjectRoleAssignments.ps1 (hunk 4 above)
Comment (SINIKI) [reply Scripts/RemediationScripts/Remediate-InvalidAADObjectRoleAssignments.ps1]: Corrected.

**Review by gvaradarajan-msft — APPROVED** [review]

**Pull request merged**